modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -0,0 +1,1229 @@
+                                             const unsigned int block_id_shift,
+                                             const bool quad_center_elements,
+                                             const unsigned int boundary_id_shift)
+{
+  unsigned int angle_number =
+      azimuthal_tangent.size() == 0 ? num_sectors_per_side : (azimuthal_tangent.size() - 1);
+  unsigned int ring_region_radial_intervals =
+      has_rings ? accumulate(rings.begin(), rings.end(), 0) : 0;
+  unsigned int duct_region_radial_intervals =
+      has_ducts ? accumulate(ducts_layers.begin(), ducts_layers.end(), 0) : 0;
+
+  unsigned int num_total_nodes = 0;
+
+  // calculate the node number of the central elements.
+  if (quad_center_elements)
+  {
+    num_total_nodes = (angle_number / 2 + 1) * (angle_number / 2 + 1) +
+                      (angle_number + 1) * (ring_region_radial_intervals + background_intervals +
+                                            duct_region_radial_intervals);
+  }
+  else
+  {
+    num_total_nodes =
+        1 + (angle_number + 1) * (ring_region_radial_intervals + background_intervals +
+                                  duct_region_radial_intervals);
+  }
+
+  std::vector<Node *> nodes(num_total_nodes); // reserve nodes pointers
+
+  // Geometries
+  const Real corner_to_corner =
+      pitch / std::cos(M_PI / side_number); // distance of bin center to cell corner
+  const Real corner_p[2][2] = {{0.0, 0.5 * corner_to_corner},
+                               {0.5 * corner_to_corner * std::sin(2.0 * M_PI / side_number),
+                                0.5 * corner_to_corner * std::cos(2.0 * M_PI / side_number)}};
+  const unsigned int div_num = angle_number / 2 + 1;
+  std::vector<std::vector<dof_id_type>> id_array(div_num, std::vector<dof_id_type>(div_num));
+  dof_id_type node_id = 0;
+  if (quad_center_elements)
+  {
+    Real ring_radii_0;
+
+    if (has_rings)
+      ring_radii_0 = ring_radii.front() / (Real)rings.front();
+    else if (has_ducts)
+      ring_radii_0 =
+          ducts_center_dist.front() * std::cos(M_PI / side_number) / (Real)background_intervals;
+    else
+      ring_radii_0 = pitch / 2.0 / (Real)background_intervals;
+    ring_radii_0 = ring_radii_0 / div_num * (div_num - 1);
+
+    mesh = center_nodes(dynamic_pointer_cast<ReplicatedMesh>(mesh),
+                        side_number,
+                        div_num,
+                        ring_radii_0,
+                        &node_id,
+                        &nodes,
+                        &id_array);
+  }
+  else
+  {
+    Point center_p = Point(0.0, 0.0, 0.0); // pin-cell center
+    nodes[0] = mesh->add_point(center_p, 0);
+  }
+
+  // create nodes for the ring regions
+  if (has_rings)
+    mesh = ring_nodes(dynamic_pointer_cast<ReplicatedMesh>(mesh),
+                      ring_radii,
+                      rings,
+                      num_sectors_per_side,
+                      &node_id,
```

Comment:
I'm concerned about node_ids overlapping because you are hard setting them to start at 0 then increase here. Is there a part where you shift them after?
What if people use several of these mesh generators?

Reply:
I think the node_ids will be automatically adjusted when the unit meshes generated by `PolygonConcentricCircleMG` are stitched by `PatternedHexMG` into an assembly mesh.

Reply:
yes in your use of these generators it is. But someone re-using that code could be surprised
@friedmud @roystgnr  what have we done in the past for meshes? Keeping track of arbitrary numbers and using arbitrary offsets seems like it could get us in trouble when combining things

Reply:
In general node/elem/unique ids are only unique to a given mesh, not unique to all meshes in a simulation run.  When libMesh stitches multiple meshes together, we always look at the maximum id values in mesh 0 to determine how far we need to offset ids in mesh 1, and so forth.  So there's always an offset involved, but it's never an arbitrary one from our perspective.

Reply:
Thank you for the detailed explanation. 
Since the onus has traditionally been on the stitcher/pattern-er object to deal with id, this is in the same vibe as the existing code base. All good!

Reply:
Thanks, @roystgnr @GiudGiud